Automattic/wp-calypso
# Themes: Disable filtering for Jetpack sites; enable deletion only for existing themes

#### Changes proposed in this Pull Request

This PR disables the unnecessary filtering of themes for Jetpack sites - this was causing any wpcom themes that exist on the site to not be added to the list of themes on the site in Redux at all.

We also enable deletion only for themes that are found on a site; if a theme isn't found, the delete button shouldn't appear at all.

An alternative to #51277 and #51300.

#### Testing instructions

* Go to `/themes/:site` where `:site` is a Jetpack site.
* Activate "Twentytwenty"
* Activate "Twentytwentyone"
* Verify you can delete Twentytwenty successfully.
* Re-check and verify you can no longer delete Twentytwenty now (the deletion button is no longer visible).
* Verify you still can't delete "Twentytwentyone".
* Verify that if you activate another theme, you can now delete "Twentytwentyone"
* Verify the themes list and deletion of other themes works the same as before.
